### ISSUE TITLE
Getting hex`s OddLength error constructing extrinsic

### ISSUE THREAD

**Issue by alekspickle** [scs/substrate-api-client#68]
Reproduce is kind of hard because of our architecture but I can privide `compose_extrinsic` macro input to check.

```
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn check_compose_extrinsics() {
        let node_url = "wss://node1-chain.akropolis.io";
        let api = Api::new(node_url.into()).set_signer(get_sr25519_pair(ANY_VALID_MNEMONIC));
        let from = H160::default(); // any eth address would suffice
        let to = AccountId32::default();
        let id = H256::default();
        let ext = compose_extrinsic!(
            api,
            "Bridge",
            "multi_signed_mint",
            id,
            from,
            GenericAddress::from(to),
            Compact(0u32),
            Compact(10000000000000000000u128)
        );
        let hash = api.send_extrinsic(ext.hex_encode(), XtStatus::Finalized);
        //                      ^^^^^^^^^^^^^^^^^---- fails here
    }
}

pub fn get_sr25519_pair(signer_mnemonic_phrase: &str) -> sr25519::Pair {
    sr25519::Pair::from_phrase(signer_mnemonic_phrase, None)
        .expect("invalid mnemonic phrase")
        .0
}
```

upd: I reproduced this as a test so it should be fairly easy

**Comment by developerfred** [comment]
@alekspickle do you want to have an error handling in the function?

**Comment by alekspickle** [comment]
@developerfred 
In theory that should not be possible, I assume?
You added an odd check which returns `""` if it is odd, but how is that even possible from compose_extrinsics macro?+

**Comment by developerfred** [comment]
@alekspickle thanks, I will update this pr

**Comment by alekspickle** [comment]
Just debugged to the point ehere I can say that this error is rising because of node error handling you use. 
`hex::decode` fails to parse "nix" that you substitute [here](https://github.com/scs/substrate-api-client/blob/23b6255502141559fef42fafd7032704c5dd1f46/src/rpc/client.rs#L126).
Since `None` value is returned for each error from node error parsing [if the value of result is neither an object nor the null](https://github.com/scs/substrate-api-client/blob/23b6255502141559fef42fafd7032704c5dd1f46/src/rpc/client.rs#L154) the result field that you can get in some cases is the reason of substitution of hexed input for `hex::decode` with `"nix"`.

I have yet to solve my issue, because I have found different `Bad input data provided to validate_transaction` in node logs, at time I am trying to send this transaction. But treat all errors like `None` and substituting it with `"nix"` is very mutch error-prone imo.
Passing the error on the other hand behaves similarly but makes debug easier.

Consider this PR #73

**Comment by brenzi** [comment]
Oh, that is really an ugly remainder of some quick hack. We will fix this....

**Comment by developerfred** [comment]
@alekspickle what do you think of this [fhler](https://docs.rs/fehler/1.0.0/fehler/) for error handling?

**Comment by brenzi** [comment]
I consider this solved by #73 and #75
